### ISSUE TITLE
Command '['/tmp/bin/rke', 'up', '--config', '/tmp/config.yaml']' returned non-zero exit status 1

### ISSUE THREAD

**Issue by risayew** [techcto/S3initRKE#2]
Something wrong with rke up.
<img width="1060" alt="screenshot 2019-01-05 at 22 55 54" src="https://user-images.githubusercontent.com/2446408/50729604-23a1e300-113d-11e9-9f25-f3349c92bf8e.png">

**Comment by techcto** [comment]
As mentioned, I kind of just stopped on this a few months back for another project, I was probably mid-way through making a change.  It looks like it was working and then died from a bug most likely.  I will do my best to check this out over the next day or so, but feel free to make a pull request if you see the issue.  There are a ton of moving parts.

**Comment by risayew** [comment]
Oh, thanks! I'm trying to find the issue too.

I see the following lines in the log(just few lines more on top than the previous one):

<img width="729" alt="screenshot 2019-01-05 at 23 12 31" src="https://user-images.githubusercontent.com/2446408/50729760-809e9880-113f-11e9-9d86-d6030b958e89.png">

**Comment by techcto** [comment]
It is supposed to make a S3 bucket in your AWS account and then upload the config file.  It looks like it is referencing my S3 bucket. Maybe I did a hard-code test.  Does it make a S3 bucket in your account?

Thank you for taking a look at this.

Do you have the S3 buckert

**Comment by techcto** [comment]
Sorry, are you using the Rancher install CloudFormation template that I made?

https://github.com/techcto/rancher-aws

Not sure why this would not be set?

[cid:image001.png@01D4A51B.998E3200]

[cid:image002.png@01D4A51B.998E3200]

**Comment by risayew** [comment]
Actually, yes, i see the created rancher-rke bucket.
<img width="310" alt="screenshot 2019-01-05 at 23 24 08" src="https://user-images.githubusercontent.com/2446408/50729844-28689600-1141-11e9-9958-8d69ea3cb026.png">

But i don't see config.yaml file in it.

**Comment by risayew** [comment]
Yes, exactly, i'm using the https://github.com/techcto/rancher-aws. 
But i am not launching the stack from github.
Instead i upload the rancher-ha-cluster.yaml.

**Comment by risayew** [comment]
Sorry, i did not get the question:
 
>>> Not sure why this would not be set?

>>> [cid:image001.png@01D4A51B.998E3200]

>>> [cid:image002.png@01D4A51B.998E3200]

**Comment by techcto** [comment]
Sorry, it looks like you are doing it right.  It must be a bug some-where.  It all worked at one time.  The base AMI’s from AWS may need updating or possibly I stopped mid-change.  I will need to try and run this again and see if it works on my end.

I will try tomorrow if I can.

**Comment by risayew** [comment]
Thank you so much!
I will work on the issue too.

**Comment by risayew** [comment]
Very strange. i just re-created the stack and it's working now.
<img width="730" alt="screenshot 2019-01-06 at 01 42 36" src="https://user-images.githubusercontent.com/2446408/50730730-64592680-1154-11e9-8028-d611a4bf4a03.png">

The only thing:
Finish manual run. This is not a Lifecycle event: Parameter validation failed:
Invalid length for parameter LifecycleHookName, value: 0, valid range: 1-inf
Invalid length for parameter LifecycleActionToken, value: 0, valid range: 36-inf

**Comment by techcto** [comment]
I think those are just logs, as I wrote the l class originally for lambda and converted to code build based on how long the process takes. Also possible to run code build manually also. I don't think those errors are bad.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by techcto** [comment]
Also the code build is supposed to run each time a servers scales in or out, it is possible this is where I stopped. Were you able to login?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by risayew** [comment]
Yes, i am able to login to rancher. everything is working.
>>> the code build is supposed to run each time a servers scales in or out

I have noticed, that just right after new install of the stack, the codebuild triggered 3 times.
<img width="1086" alt="screenshot 2019-01-06 at 17 09 56" src="https://user-images.githubusercontent.com/2446408/50738601-ff461500-11d5-11e9-9229-5f48e68b595a.png">
The strange thing here is, that the build time was short for the "later" triggered builds. So the first triggered build was still working after later triggered builds did already finish. Actually,  the "Command '['/tmp/bin/rke', 'up', '--config', '/tmp/config.yaml']' returned non-zero exit status 1" did happen exactly in this first long build job.

**Comment by techcto** [comment]
I think this is where I stopped, I was not able to properly control the scaling events and possibly the code was being ran too many times.  Or at the wrong time. Do you have experience with AWS?  If so, please feel free to make a pull request and help finish this.  I have been planning on getting back on this project once I get time.  Any help is much appreciated.

**Comment by risayew** [comment]
I've just started learning aws, but definitely i will help to solve the problem.
Thank you for the templates you shared!
